### ISSUE TITLE
Pull internal logs via control Graphql

### ISSUE THREAD

**Issue by georgeee** [MinaProtocol/mina#13064]
Internal logs are used for VS' dashboard and take the following form in Ocaml codebase:

```
[%log internal] "Update_ledger_and_get_statements" ~metadata:[ ("partition", `String "single") ]
```

We need to come up with:

* Control Graphql query that would allow to pull logs starting from some index
* Control Graphql mutation that would allow to clean logs up until some index

Logs are to be stored in a bounded queue with old logs to be overwritten if not queried before that. "Bound" is to be configured via application config/CLI flag.

Feature is to be injected via setting up logging with a new backend when the `-itn-keys`/`--itn-port` flags are provided.

**Comment by georgeee** [comment]
TODO: come up with query/mutation format

**Comment by psteckler** [comment]
Query could be `internalLogs`. Input could request the most recent `n` logs, or logs starting at a timestamp. If those are omitted, provide all available logs.

The logger outputs a sprintf-style interpolated string, and the "metadata", a list of string, JSON pairs. In GraphQL, each log entry can be a pair consisting of the interpolated string, and a list of string, string pairs.

The mutation could be `flushInternalLogs`, which empties the bounded queue.

The CLI can take a new optional flag `--itn-max-internal-logs`.

**Comment by georgeee** [comment]
Graphql schema (draft):


```
type query {
internalLogs(
    startLogId : Int!,
    limit : Int!
  ): String!
}

type mutation {
flushInternalLogs(
    endLogId: Int!
  ): Bool!
}
```

Each log should be assigned a unique integer log id. Return of `internalLogs` contains logs ordered by log id concatenated with `\n`.
Query `internalLogs` returns logs with ids greater or equal to `startLogId`. It returns up to `limit` logs, where `limit` is taken from `--itn-max-internal-logs` CLI argument.

Mutation `flushInternalLogs` removes logs with ids up until `endLogId` inclusive.

**Comment by psteckler** [comment]
Here's what my first attempt at implementation returns for the logs:
```
{
  "data": {
    "internalLogs": [
      {
        "id": 10,
        "message": "Pickles__Step_main:File \"src/lib/pickles/step_main.ml\", line 525, characters 8-23 Step_compute_bulletproof_challenges_done []"
      },
      {
        "id": 11,
        "message": "Pickles__Step_main:File \"src/lib/pickles/step_main.ml\", line 390, characters 8-23 Step_compute_bulletproof_challenges []"
      },
      {
        "id": 12,
        "message": "Pickles__Step_main:File \"src/lib/pickles/step_main.ml\", line 525, characters 8-23 Step_compute_bulletproof_challenges_done []"
      },
      {
        "id": 13,
        "message": "Pickles__Step_main:File \"src/lib/pickles/step_main.ml\", line 390, characters 8-23 Step_compute_bulletproof_challenges []"
      },
      {
        "id": 14,
        "message": "Pickles__Step_main:File \"src/lib/pickles/step_main.ml\", line 525, characters 8-23 Step_compute_bulletproof_challenges_done []"
      }
]
}
```
It's returning a list of int, string pairs, not concatenated strings. The `[]`s are the metadata (empty for all these logs).
The start and end ids are just the ones at the beginning and end of the response, so maybe redundant to give those explicitly.

For `flushQueue`, it's returning the number of deleted items, which is more informative than `true`.

**Comment by psteckler** [comment]
Here's a log with a nonempty `metadata` part:
```
      {
        "id": 272,
        "message": "Verifier__Prod:File \"src/lib/verifier/prod.ml\", line 622, characters 2-17 Verify_transaction_snarks [count:2]"
      }
```

**Comment by psteckler** [comment]
Closed by #13084.
